common/achievement/index.ts

```diff
@@ -62,10 +82,59 @@ async function trackEvent(event: ActionEvent, context: NotificationContext) {
                 value: value,
                 action: event.actionId,
                 userId: event.user.userID,
-                relation: getRelationByContext(context),
+                // TODO - get relation OR get relationId from context
+                relation: event.context.relationId ?? '',
             },
         });
     }
 
     return true;
 }
+
+async function checkUserAchievement(userAchievement: UserAchievementTemplate | undefined) {
+    if (userAchievement) {
+        const evaluationResult = await isAchievementConditionMet(userAchievement);
+        if (evaluationResult.conditionIsMet) {
+            const dataAggregationKey = Object.keys(userAchievement.template.conditionDataAggregations as ConditionDataAggregations)[0];
+            const evaluationResultValue =
+                typeof evaluationResult.resultObject[dataAggregationKey] === 'number' ? Number(evaluationResult.resultObject[dataAggregationKey]) : null;
+            const awardedAchievement = await awardUser(evaluationResultValue, userAchievement);
+            const userAchievementContext: UserAchievementContext = {};
+            await createAchievement(awardedAchievement.template, userAchievement.userId, userAchievementContext);
+        }
+    }
+}
+
+async function isAchievementConditionMet(achievement: UserAchievementTemplate) {
+    const {
+        userId,
+        template: { condition, conditionDataAggregations, metrics },
+    } = achievement;
+    if (!condition) {
+        return;
+    }
+
+    const updatedCondition = injectRecordValue(condition, achievement.recordValue);
```

Comment:
the recordValue is the value of the highest streak yet achieved by a user. this has to be stored in a user achievement and not the template, So it can not be put in the condition data. We do not try to find the highest streak ever achieved on the run but store it with the achievement and update it when achieved.
putting it in the result object is not very useful since the evaluation is done based on the condition, that has to contain the actual value, not a variable.